client/editorCtrl.ts

```diff
@@ -87,8 +87,28 @@ export class EditorController {
                 deleteOnDropOff: true,
             },
             addDimensionsCssVars: true,
+
+            pocketRoles: (color: cg.Color):string[] | undefined=>{return this.variant.pocketRoles(color);},
         });
 
+        //
+        ['mouseup', 'touchend'].forEach(name =>
+            [this.chessground.state.dom.elements.pocketTop, this.chessground.state.dom.elements.pocketBottom].forEach(pocketEl => {
+                if (pocketEl) pocketEl.addEventListener(name, (e: cg.MouchEvent) => {
+                    this.dropOnPocket(this.chessground.state, e);
+                } )
+            })
+        );
+        eventsDragging.forEach(name =>
+            [this.chessground.state.dom.elements.pocketTop, this.chessground.state.dom.elements.pocketBottom].forEach(pocketEl => {
+                if (pocketEl) pocketEl?.childNodes.forEach(p => {
+                    p.addEventListener(name, (e: cg.MouchEvent) => {
+                    this.drag(this.chessground.state, e);
+                } ) });
+            })
+        );
+
+        //
```

Comment:
Yeah i don't know. 

It was easier to leave this code in CG together with the rest of the event handling that got moved there, but I decided intentionally to split it and move it back to pychess, just because it doesn't sound like something that is part of the usual board dynamics as part of a game.

But for sure it is murky and not clear cut how to think about these things and how to conceptualize them.
For example "free" mode can just as well be interpreted as allowing move from board to pocket as it allows moving a pawn on 1st rank. 

Ultimately my "philosophical razor" was something like this: "if something is useful ONLY for editor but not for any game - move it out of CG".


If there was at least one variant that actually needs dropping pieces inside a pocket it would make this question much easier to think about and going in direction of making it part of CG. 

Another place i used this "razor" was when deciding if we should allow for arbitrary count and configuration of pockets or just hard-code to 2 (top and bottom). In editor mode we have also those 2 "palettes" pockets in addition to the usual "game pockets" and it is possible to also have those as CG pockets if CG supported arbitrary pocket combinations, but again it makes sense only for editor and not any game variant so I went against such idea. If there was a variant that has more than one pocket per side then it would have been different - instead of top/bottom properties for elements (and per-color in the state) would have had some variable length lists and then editor could take advantage of that.